### DISCUSSION TITLE
Easy docker commands to setup?

### DISCUSSION THREAD

**Discussion by seanwatkins** [waschinski/photo-stream#39] [General]
Does anyone have any easy commands to get this going under docker?? I'm setting it back up, and no luck.

I get these errors:

      
Could not locate Gemfile or .bundle/ directory

Could not locate Gemfile or .bundle/ directory

**Comment by waschinski** [comment]
Hi, can you be a little bit more detailed on what did you try so far that resulted in these errors?

Easiest way should be using the pre-build docker images on [Docker Hub](https://hub.docker.com/r/waschinski/photo-stream). Are you using Docker CLI or some GUI tool like Portainer?

**Reply by seanwatkins** [reply]
Docker CLI + Portainer.

I did it to go -

Where it was jamming up --- the photos directory.

So you need to mount your photos into /photo-stream/photos

And I originally had all my settings in /photo-stream/photos/.env

I copied the .env file, to /photo-stream

Is there a better way todo this, so it always works on re-create?

**Reply by waschinski** [reply]
> Docker CLI + Portainer.
> 
> Is there a better way todo this, so it always works on re-create?

Just use environment variables using `docker compose run -e` or set them in Portainer instead of using the `.env` file. You can also set up a stack for photo-stream in Portainer, put the content of the compose file there and adjusting the env variables.

That `.env` file is better to be used when running photo-stream locally.

**Reply by seanwatkins** [reply]
Ok good suggestion! Maybe we should put some examples of doing this in the Readme?

BTW, is it generally safe, to use _site as a root of a website?

**Reply by waschinski** [reply]
If you have an idea on how to improve the Readme feel free to suggest something here or open up a PR. I am all for making it easier for others to use photo-stream!

May I return the question: what makes you think it's not safe to use _site?

**Reply by seanwatkins** [reply]
I see the scripts inside to rsync it out... better to use _site as a volume, or the scripts?

On another topic, now my container won't run.

``

W, [2022-09-08T17:34:00.871127 #1]  WARN -- : Badly formed IFD: Infinity


W, [2022-09-08T17:34:00.871731 #1]  WARN -- : Badly formed IFD: Infinity


  Liquid Exception: No such file or directory @ rb_sysopen - photos/original/LIBF8188-original-65ba1d.jpeg in feed.xml


                    ------------------------------------------------


      Jekyll 4.2.2   Please append `--trace` to the `serve` command 


                     for any additional information or backtrace. 


                    ------------------------------------------------

``
I deleted the file, (Maybe bad jpeg) but now, its like it has cached this file...  it seems like I have to nuke the entire docklet to get it to go again? That file name above,  the LIBF8188-original -- its gone, but its finding the cached file somewhere?

**Reply by waschinski** [reply]
> I see the scripts inside to rsync it out... better to use _site as a volume, or the scripts?

Jekyll is creating all the static files in the `_site` folder and the web server is serving only that folder. `_scripts` is not accessible from outside.

> On another topic, now my container won't run.
> W, [2022-09-08T17:34:00.871127 #1] WARN -- : Badly formed IFD: Infinity

No idea about that error message to be honest.

**Comment by seanwatkins** [comment]
2 quick things

1 - I got it all going... so its sitting listing on port 4000. If I drop more pictures in, does it figure that out, and re-gen? Or do I need to restart the container each time?

2 - Here is my actual commands for going now... I'm no worthsmith!

```
docker run -p 4000:4000 \
       --rm \
       --name photo-stream \
       -v /share/BIG/Pictures/photo-stream/:/photo-stream/photos \
       -v /share/BIG/Pictures/_site/:/photo-stream/_site \
       -w /photo-stream \
       -e TITLE="SW Photos" \
       -e AUTHOR_NAME="Sean Watkins -e AUTHOR_EMAIL="sean.watkins@gmail.com \
       -e URL="https://p.1349175ab.com" \
       -e SHOW_OFFICIAL_GITHUB=0 \
       -e ALLOW_ORDER_SORT_CHANGE=1 \
       -e ALLOW_ORIGINAL_DOWNLOAD=1 \
       -e PHOTO_PATH=./photos \
       -e INSTAGRAM_USERNAME=seanwatkins3165 \
       -e CUSTOM_LINK_NAME=STRAVA \
       -e CUSTOM_LINK_URL="https://www.strava.com/athletes/35611001" \
waschinski/photo-stream:latest
```

**Reply by waschinski** [reply]
> If I drop more pictures in, does it figure that out, and re-gen?

Yup. It does rebuild everything automatically. 

>        -v /share/BIG/Pictures/_site/:/photo-stream/_site \

What's the reason for doing this though?

**Reply by seanwatkins** [reply]
Should it find them in real time, or it wake up every say 3600 seconds or something?

Just posting those commands in there, so you can put in the README for real Quickstart.

I had a VM running on Azure for this, but the charges... forget it. In the process of setting this back up home, I had to start from scratch.. there wasn't any exact example for docker cli.

Hope it helps.

**Reply by waschinski** [reply]
> Should it find them in real time, or it wake up every say 3600 seconds or something?

Real time, but processing and building the site can take a a little time if you have lots of photos.

> I had a VM running on Azure for this, but the charges... forget it. In the process of setting this back up home, I had to start from scratch.. there wasn't any exact example for docker cli.

I see. But there's actually no reason to map the `_site` folder to the host. The site is accessible on the defined port and if you want your photo-stream to be available on a domain you will have to set it up accordingly.

**Comment by seanwatkins** [comment]
Sorry I didn't totally see what you meant with the -v  thats just the share I have at home, mapped into the container.
